### PR TITLE
Enemy attack interval setting

### PR DESCRIPTION
**Pull request by kattalainen**
Ai component now has an attack interval variable so rate of attacks can be regulated. Added argument for enemyfactory to set the interval.

### CODE REVIEW

File: StortSpelProjekt/Game/src/Gamefiles/Components/AiComponent.cpp [hunk 1]
```diff
@@ -68,9 +70,20 @@ void component::AiComponent::Update(double dt)
...
+						Log::Print("ENEMY ATTACK!\n");
+						m_TimeAccumulator = 0.0;
+					}
 				}
 			}
+			else
+			{
+				// "reset" accumulator if out of attack range
+				m_TimeAccumulator = 0.0;
+			}
```
Comment (Schtekt) [inline StortSpelProjekt/Game/src/Gamefiles/Components/AiComponent.cpp:142]: The cooldown should not be reset every time the enemy is out of range. It should be reset once the enemy has attacked.
For example, the enemy is outside attack range for a large amount of time, then attacks whenever the target is within range. There the cooldown starts instead.

**Review by fjz345 — DISMISSED** [review]
Looks good.

**Review by Schtekt — CHANGES_REQUESTED** [review]
I would like to see that "m_TimeAccumulator" only is reset after an attack has been made.
If for example, an enemy with a large cooldown (for a BIG attack) needs to wait while being inside attack range all the time, that would not be too good now, would it?

File: StortSpelProjekt/Game/src/Gamefiles/Components/AiComponent.cpp (hunk 1 above)
Comment (kattalainen) [reply StortSpelProjekt/Game/src/Gamefiles/Components/AiComponent.cpp:142]: I disagree, fight me

**Review by Schtekt — DISMISSED** [review]

**Review by Schtekt — APPROVED** [review]

**Review by fjz345 — APPROVED** [review]

**Pull request merged**